src/lib/server/orders.ts

```diff
@@ -639,6 +670,49 @@ export async function createOrder(
 
 	for (const item of items) {
 		item.discountPercentage ??= discountByProductId.get(item.product._id) ?? wholeDiscount;
+		if (item.freeQuantity) {
+			let quantityToConsume = Math.min(item.quantity, item.freeQuantity);
+			const freeProductSubscriptions = await collections.paidSubscriptions
+				.find({
+					[`freeProductsById.${item.product._id}`]: { $exists: true },
+					[`freeProductsById.${item.product._id}.available`]: { $gt: 0 },
```

Comment:
```suggestion
					[`freeProductsById.${item.product._id}.available`]: { $gt: 0 },
```

you should also restrict the subscriptoin to the user making the order, not all subscriptoins